### PR TITLE
The Pickup Macro is made null for guns.

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

"Dude why not just get a pick-up macro for your guns? why bother with the mag harness?"

## Why It's Good For The Game

```pick-up HG-37-12-pump-shotgun\nPick-up T-35-pump-shotgun\nPick-up M240-T-incinerator-unit\nPick-up GAL9-submachinegun\nPick-up T-35-pump-shotgun\nPick-up MBX-900-Lever-Action-Shotgun\nPick-up ZX-76-assault-shotgun\nPick-up (Heavy-Stubber)\nPick-up MK221-tactical-shotgun\nPick-up m37a2-pump-shotgun\nPick-up m4ra-battle-rifle\nPick-up m41ae2-heavy-pulse-rifle\nPick-up m43-sunfury-lasgun-mk1\nPick-up m41a1-pulse-rifle\nPick-up m2132-machete\nPick-up  m39-submachinegun\nPick-up double-barrel-shotgun\nPick-up 88-mod-4-combat-pistol\nPick-up m5-'night-raider'-survival-knife\nPick-up sawn-off-shotgun\nPick-up knife\nPick-up m11-throwing-knife\nPick-up m4a3-service-pistol\nPick-up m44-combat-revolver\nPick-up Paladin-12-pump-shotgun\nPick-up custom-built-shotgun\nPick-up M16-rifle\nPick-up Ceremonial-Sword\nPick-up M1911A1-custom-pistol\nPick-up T-64-designated-marksman-rifle\nPick-up T-18-Carbine\nPick-up FN-M16A4\nPick-up TP-44-combat-revolver \nPick-up M41A-pulse-rifle\nPick-up M240A1-incinerator-unit```

Ya see this above? This is the particular thing I am destroying.

If you want to retain your weapon, put a mag harness on it.

## Changelog
:cl: Hughgent
tweak: The pick-up macro will now fail for guns, strap a mag harness on it to keep it with you.
/:cl:

### CODE REVIEW

**Comment by David-Stormwell** [comment]
I'm sueing for stealing my marco

**Comment by Pariah919** [comment]
I don't personally like this. This isn't a good change.

**Comment by jroinc** [comment]
I like this and think it's a good change. Fuck macros, never used them.

Issue is, I think you can just have it target ALL guns per serial number.

CM had an old, OLD issue where you could have a macro to, say, "click on young runner 1-999" and it would do it.

**Comment by SilencedMP5A5** [comment]
mmmm don't like

**Comment by Wayland-Smithy** [comment]
> you could have a macro to, say, "click on young runner 1-999" and it would do it.

Hopefully we didn't inherit that flaw, but at any rate the possibility of it has me concerned this will just become an arms race. 

Would it be possible to just ignore Pick-up's that do not come from mouse clicks?

**Comment by SplinterGP** [comment]
i dont care about this change but picking up weapon macro aint just same as literally clicking the gun ?

**Comment by David-Stormwell** [comment]
> 
> 
> i dont care about this change but picking up weapon macro aint just same as literally clicking the gun ?

No, I mostly used it in events where I was A: too busy to pixel hunt for a gun which is either under a item or under a xeno sprite, or B: on the run while I need my mouse somewhere else.
This isn't a extreme nerf, but if you are a frontline marine who gets stunned quite some then you are shit outa luck in some events.

**Comment by SilencedMP5A5** [comment]
xeno mains detected...

**Comment by Dolth** [comment]
Yup, totally not like it.
Mag harness prevent you from losing your gun at all when flinged by a warrior or shrike or whatever, you keep it after being captured.
The macro prevent from spamming click on the floor and missing miserably.
Also since I play on CM (2015, same gamemode) I never had anyone complain about the pick-up macro.
Big nono imo.

**Comment by Hughgent** [comment]
My argument is this.
1. Make the mag harness more of an important choice.
2. when you drop something on the ground, it SHOULD be a scramble to pick it up. This replicates a person also hunting on the ground for some item during a tense moment of action.

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/castedatum_hivelord.dm [hunk 1]
```diff
@@ -66,7 +66,7 @@
 		/datum/action/xeno_action/activable/xeno_spit,
 		/datum/action/xeno_action/create_jelly,
 		/datum/action/xeno_action/place_jelly_pod,
-		)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/castedatum_hivelord.dm:69]: atomize

**Comment by TiviPlus** [comment]
Also I dont like the serial numbers and still think we should do my randomized extra spaces idea

**Comment by Gooboi** [comment]
I dunno, pick-up macros are kind of "eh" since some guns are so huge it's easy to just click on them anyway, it seems the only time a pick-up macro might ever even be useful is when you're being cornered by xenos and dropped a gun below one of them that also happen to be standing on a pile of other items... 

That being said, would this change even work? If all that's going on is putting serial numbers on weapons, people can easily just modify their macro round-start to match their weapon serial number.

**Comment by Hughgent** [comment]
Modifying their pickup macro is perfectly fine. This just stops the 'set it and forget it' style of macro. It is perfectly acceptable to become 'overly familiar' with your personal weapon, knowing how it falls, how to pick it up. The assumption is that this will become annoying to do and only the sweatiest of nerds will do this every single round.

This is because `pick-up` is based on the name of the object. Applying a random number to the name makes it 'different' than the stored macro.

**Comment by Dolth** [comment]
When your gun is under the sprite of a queen and some acid sprays or a crusher and you need to right click, the pick up macro is just mandatory.

If you randomize a serial number or something to force people to macro at the end of each round, then I don't see any issue anymore. (Altho I presume people won't like it)

**Comment by Hughgent** [comment]
Counter point.

When a giant space alien is on top of your weapon, it should be difficult to pickup. Put the mag harness on your weapon to prevent this.

**Comment by SilencedMP5A5** [comment]
this really seems like a salt pr ngl

i don't use macros like that btw

**Comment by davidchan13** [comment]
I don't even use pick up macros and this seems like salt PR. It's all the arguments against bump slashing.

**Comment by Chaznoodles** [comment]
Pickup macros bad

This change goodf

**Comment by Knuff-les** [comment]
This doesn't make any sense, Cause using that same logic you would submit the same thing with all the pistols, cause you can do it with ALL the pistols. +1 to the salt PR idea

**Review by spookydonut — CHANGES_REQUESTED** [review]
just stop the pickup verb from working on guns

**Pull request merged**